File: shared/contracts/analytics.ts
export type TotalCostOfOwnership = {
  manual: number
  orchestrated: number
  delta: number
}

export type DashboardMetrics = {
  median_time_to_cash_hours: number
  guardrail_compliance_rate: number
  failure_auto_recovery_rate: number
  cost_per_100_deals: number
  manual_vs_orchestrated_tco: TotalCostOfOwnership
}

File: server/app/schemas/analytics.py
from app.schemas.common import ORMModel


class TotalCostOfOwnership(ORMModel):
    manual: float
    orchestrated: float
    delta: float


class DashboardMetrics(ORMModel):
    median_time_to_cash_hours: float
    guardrail_compliance_rate: float
    failure_auto_recovery_rate: float
    cost_per_100_deals: float
    manual_vs_orchestrated_tco: TotalCostOfOwnership
